Userland/Applications/Calendar/Calendar.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Calendar.h"
+#include <AK/JsonParser.h>
+#include <LibConfig/Client.h>
+#include <LibFileSystemAccessClient/Client.h>
+
+namespace Calendar {
+
+Calendar::Calendar()
+{
+    m_events = AK::JsonArray();
+}
+
+OwnPtr<Calendar> Calendar::create()
+{
+    return adopt_own(*new Calendar());
+}
+
+ErrorOr<void> Calendar::add_event(JsonObject event)
+{
+    TRY(m_events.append(move(event)));
+
+    set_dirty(true);
+
+    on_events_change();
+
+    return {};
+}
+
+void Calendar::set_events(JsonArray events)
+{
+    m_events = move(events);
+    on_events_change();
+}
+
+ErrorOr<void> Calendar::save(FileSystemAccessClient::File& file)
+{
+    set_filename(file.filename());
+    set_dirty(false);
+
+    auto stream = file.release_stream();
+    TRY(stream->write_some(m_events.to_deprecated_string().bytes()));
+
+    stream->close();
+
+    return {};
+}
+
+ErrorOr<void> Calendar::load_file(FileSystemAccessClient::File& file)
+{
+    set_filename(file.filename());
+    set_dirty(false);
+
+    auto content = TRY(file.stream().read_until_eof());
+    auto events = TRY(AK::JsonParser(content).parse());
+
+    // TODO: Alert user that calendar file is invalid
```

Comment:
Is it trivial to do this in this PR?

Reply:
iʼm not sure thereʼs a validator for the whole structure (itʼd be nice to check it here); thatʼs a neat “yak”, i guess, to have e.g. a json schema validator. 😉